sdcm/logcollector.py

```diff
@@ -1134,6 +1136,34 @@ def is_collect_to_a_single_archive(self) -> bool:
         return True
 
 
+class SSTablesCollector(SCTLogCollector):
+    """
+    Collect corrupted sstables from db nodes.
+    """
+    log_entities = [
+        FileLog(name='parallel-timelines-report.html',
+                search_locally=True)
+    ]
+    cluster_log_type = "corrupted-sstables"
+    cluster_dir_prefix = "corrupted-sstables"
+
+    def collect_logs(self, local_search_path: Optional[str] = None) -> list[str]:
+        s3_links = []
+        raw_events_file_path = Path(self.local_dir).parent.parent.parent / EVENTS_LOG_DIR / RAW_EVENTS_LOG
+        with open(raw_events_file_path, "r", encoding="utf-8") as events_file:
+            node_names = {event.get("node") for raw_line in events_file.readlines()
+                          if (event := json.loads(raw_line)).get("type") == "MALFORMED_SSTABLE"}
+        nodes_with_malformed_sstable: List[CollectingNode] = [node for node in self.nodes if node.name in node_names]
+        for node in nodes_with_malformed_sstable:
+            LOGGER.info("Collecting sstables for node %s...", node.name)
+            node.remoter.sudo("systemctl stop scylla-server.service")
+            node.remoter.sudo("pkill -9 scylla")
+            s3_links.append(upload_remote_dir_to_s3(node.ssh_login_info, "/var/lib/scylla/data", s3_bucket=S3Storage.bucket_name,
```

Comment:
are you uploading all the data directory to S3?
i thought we are going to upload only the corrupted one..

Reply:
@roydahan WDYT? Shall we copy only corrupted ones or everything?

Reply:
We agreed with @roydahan on a call, we're going to publish all the sstables when they don't exceed 400GB, otherwise we collect only ones listed in error events + 100 most recent ones (should be small)